packages/upload-api/src/ucan/conclude.js

```diff
@@ -1,106 +1,34 @@
-    const httpPutTaskGetRes = await tasksStorage.get(ranInvocation.link())
-    if (httpPutTaskGetRes.error) {
-      if (httpPutTaskGetRes.error.name === 'RecordNotFound') {
-        return {
-          error: new ReferencedInvocationNotFound(ranInvocation.link()),
-        }
-      }
-      return httpPutTaskGetRes
-    }
-
-    // Store receipt
-    const receiptPutRes = await receiptsStorage.put(receipt)
-    if (receiptPutRes.error) {
-      return {
-        error: receiptPutRes.error,
-      }
-    }
-
-    // THIS IS A TEMPORARY HACK
-    // Schedule `blob/accept` if there is a `http/put` capabilities
-    // inside the invocation that this receipt comes from
-    const scheduleRes = await Promise.all(
-      httpPutTaskGetRes.ok.capabilities
-        // Go through invocation tasks and get all `http/put`
-        .filter((cap) => cap.can === HTTP.put.can)
-        // @ts-expect-error body exists in `http/put` but unknown type here
-        .map(async (/** @type {API.HTTPPut} */ cap) => {
-          // Get triggering task (blob/allocate) by checking blocking task from `url`
-          /** @type {API.UnknownLink} */
-          // @ts-expect-error ts does not know how to get this
-          const [, blobAllocateTaskCid] = cap.nb.url['ucan/await']
-          const blobAllocateTaskGet = await tasksStorage.get(
-            blobAllocateTaskCid
-          )
-          if (blobAllocateTaskGet.error) {
-            return blobAllocateTaskGet
-          }
-
-          /** @type {API.BlobAllocate} */
-          // @ts-expect-error ts does not know how to get this
-          const allocateCapability = blobAllocateTaskGet.ok.capabilities.find(
-            // @ts-expect-error ts does not know how to get this
-            (/** @type {API.BlobAllocate} */ allocateCap) =>
-              equals(allocateCap.nb.blob.digest, cap.nb.body.digest) &&
-              allocateCap.can === W3sBlob.allocate.can
-          )
-
-          const blobAccept = await W3sBlob.accept
-            .invoke({
-              issuer: id,
-              audience: id,
-              with: id.did(),
-              nb: {
-                blob: cap.nb.body,
-                space: allocateCapability.nb.space,
-                _put: {
-                  'ucan/await': ['.out.ok', ranInvocation.link()],
-                },
-              },
-              // Expiry is set to `Infinity` so that CID will come out the same
-              // as returned in effect on `blob/add`
-              expiration: Infinity,
-            })
-            .delegate()
-
-          return tasksScheduler.schedule(blobAccept)
-        })
+    // 🚧 THIS IS A TEMPORARY HACK 🚧
+    // When we receive a receipt for the invocation we want to resume the tasks
+    // that were awaiting in the background. In the future task scheduler is
+    // expected to handle coordination of tasks and their dependencies. In the
+    // meantime we poll `blob/allocate` tasks that were awaiting for the
+    // `http/put` receipt.
```

Comment:
nit: if we need to add further triggers like this in the future, we would need to decouple this into where it is. There is why I would check on it, and I still think it is better to do

Reply:
I'm not sure I follow what you are saying here. Are you saying what if we need to resume some other blocked task ? If so I think answer there would to add something like `Scheduler.poll(....)` which would go and poll each task we need to poll that way logic around individual tasks lives where those task are defined as opposed to been entangled with conclude handler with a branched logic